test/services/database/challenge_repository_test.dart

```diff
@@ -213,23 +218,92 @@ void main() {
   });
 
   group("Challenge completion", () {
-    test("Complete a challenge", () async {
+    test("Complete a valid challenge", () async {
       FirestorePostGenerator generator = FirestorePostGenerator();
       await generator.addPostsReturnDataOnly(firestore, inChallengeRange, 1);
       final challenges = await challengeRepository.getChallenges(uid, userPos);
       expect(challenges.length, 1);
 
+      await setUserFirestore(firestore, testingUserFirestore);
       final challenge = challenges.first;
       expect(challenge.data.isCompleted, false);
-      await challengeRepository.completeChallenge(
+      bool isComplete = await challengeRepository.completeChallenge(
         uid,
         challenge.postId,
       );
 
+      expect(isComplete, true);
+
       final updatedChallenges =
           await challengeRepository.getChallenges(uid, userPos);
       expect(updatedChallenges.length, 1);
       expect(updatedChallenges.first.data.isCompleted, true);
+
+      final updatedUser = await userRepository.getUser(uid);
+      expect(
+        updatedUser.data.centauriPoints,
+        ChallengeRepositoryService.soloChallengeReward,
+      );
+    });
+
+    test("Complete a challenge that does not exist", () async {
```

Comment:
```suggestion
    test("Can't complete a challenge that does not exist", () async {
```
Little suggestion to clarify the outcome :)
It does become obvious once the code is read, but I was a little unsure about what to expect at first.
Maybe (instead?) you could add a little explanation on the expected outcomes.
This suggestion applies to the other tests of the file

Reply:
I agree, done in 64587f42e5d3fc82e416243e4222cced229d1d35